Airwallex/ApplePay/AWXApplePayProvider.m

```diff
@@ -25,26 +25,55 @@ @interface AWXApplePayProvider ()<PKPaymentAuthorizationControllerDelegate>
 @property (nonatomic, strong, nullable) AWXConfirmPaymentIntentResponse *lastResponse;
 @property (nonatomic, strong, nullable) NSError *lastError;
 @property (nonatomic) BOOL shouldInvokeCompleteWithResponse;
+@property (nonatomic) BOOL isApplePayInitiatedDirectly;
 
 @end
 
 @implementation AWXApplePayProvider
 
-#pragma mark - AWXDefaultProvider parent methods
+#pragma mark - Initiate Apple pay flow
+- (void)makePayment {
+    NSString *errorReason = @"";
+    if ([AWXApplePayProvider canHandleSession:self.session errorReason:&errorReason]) {
+        _isApplePayInitiatedDirectly = true;
+        [self handleFlow];
+    } else {
+        NSError *error = [NSError errorWithDomain:AWXSDKErrorDomain
+                                             code:-1
+                                         userInfo:@{NSLocalizedDescriptionKey: errorReason}];
+        [[self delegate] provider:self didCompleteWithStatus:AirwallexPaymentStatusFailure error:error];
+    }
+}
 
-+ (BOOL)canHandleSession:(AWXSession *)session paymentMethod:(AWXPaymentMethodType *)paymentMethod {
++ (BOOL)canHandleSession:(AWXSession *)session errorReason:(NSString *_Nullable *)error {
     if ([session isKindOfClass:[AWXOneOffSession class]]) {
         AWXOneOffSession *oneOffSession = (AWXOneOffSession *)session;
         if (oneOffSession.applePayOptions == nil) {
+            if (error) {
+                *error = NSLocalizedString(@"Missing Apple Pay options in session.", nil);
+            }
             return NO;
         }
-        return [PKPaymentAuthorizationController canMakePaymentsUsingNetworks:AWXApplePaySupportedNetworks()
-                                                                 capabilities:oneOffSession.applePayOptions.merchantCapabilities];
+        BOOL canMakePaymentUsingNetworksAndCapabilities = [PKPaymentAuthorizationController canMakePaymentsUsingNetworks:AWXApplePaySupportedNetworks()
```

Comment:
It seems on iOS 15+, Apple Pay can be displayed even though there are no cards because Apple added the ability for customers to add cards in the payment sheet (see WWDC '21 "What's new in Wallet and Apple Pay"), so I wonder if there're no cards added in the Apple Wallet yet we still want to pop the sheet anyway for iOS 15+ implementation.

Reply:
This is already an existing check, do we want to remove checking for SupportedNetworks and just check for `canMakePayments()`?

Reply:
I guess we need to test for iOS 15+ (i.e. still keep this existing logic for older versions):

- Does canMakePayments() return `true` even I didn't add card to wallet?
- Does canMakePayments(supportedNetworks) return `false` if I didn't add card to wallet?

If both answers are yes, then we should just use `canMakePayments()` for iOS 15+.

Reply:
Good catch on this one. I tested on iOS 15.0+ on device without a card and `canMakePayments` does allow to present the apple pay sheet where as  `canMakePayments(supportedNetworks)` does not and aligns with what they have mentioned on the doc also. I have updated the implementation based on your suggestion.